src/wayland.rs

```diff
@@ -569,7 +570,14 @@ impl Dispatch<wl_buffer::WlBuffer, ()> for AppData {
 
 // Connect to wayland and start task reading events from socket
 pub fn connect_to_wayland() -> wayland_client::Connection {
-    wayland_client::Connection::connect_to_env().unwrap_or_else(|err| {
+    let portal_socket = std::env::var("PORTAL_WAYLAND_SOCKET")
```

Comment:
I believe if `xdg-desktop-portal` ends up spawning a subprocess for any reason, this fd won't be parked `CLOEXEC` and will be passed to the new processs.

Probably best to add a call to set `CLOEXEC`. (Like `wayland-client` and `libwayland` do with `WAYLAND_SOCKET`.)

Reply:
Ok, I'lll add that, thanks.